ReplayKitExample/ReplayKitExample/ReplayKitVideoSource.swift

```diff
@@ -228,8 +253,29 @@ class ReplayKitVideoSource: NSObject, VideoSource {
         var videoOrientation = VideoOrientation.up
         if let sampleOrientation = CMGetAttachment(sampleBuffer, key: RPVideoSampleOrientationKey as CFString, attachmentModeOut: nil),
             let coreSampleOrientation = sampleOrientation.uint32Value {
+            let cgImageOrientation = CGImagePropertyOrientation(rawValue: coreSampleOrientation)
             videoOrientation
-                = ReplayKitVideoSource.imageOrientationToVideoOrientation(imageOrientation: CGImagePropertyOrientation(rawValue: coreSampleOrientation)!)
+                = ReplayKitVideoSource.imageOrientationToVideoOrientation(imageOrientation: cgImageOrientation!)
+
+            // TODO: Derive scaled size from format request.
+            if let adapter = vsyncInputAdapter {
+                if let adapted = adapter.scale(input: sourcePixelBuffer, maxWidthOrHeight: ReplayKitVideoSource.kDownScaledMaxWidthOrHeightSimulcast) {
+                    sourcePixelBuffer = adapted
+                } else {
+                    print("CE: Dropping frame due to buffer already in flight!")
+                    return
+                }
+            }
+        } else {
+            // TODO: Derive scaled size from format request.
+            if let adapter = vsyncInputAdapter {
+                if let adapted = adapter.scale(input: sourcePixelBuffer, maxWidthOrHeight: ReplayKitVideoSource.kDownScaledMaxWidthOrHeightSimulcastApp) {
+                    sourcePixelBuffer = adapted
+                } else {
+                    print("CE: Dropping frame due to buffer already in flight!")
+                    return
+                }
+            }
         }
```

Comment:
Move scaling out of if/else unless we are keeping CoreImage rotations. Just the scaling is enough to fix the vsync issues.